db/db_impl.cc

```diff
@@ -1011,6 +1011,20 @@ void DBImpl::PurgeObsoleteFiles(const JobContext& state, bool schedule_only) {
       case kLogFile:
         keep = ((number >= state.log_number) ||
                 (number == state.prev_log_number));
+
+        /* We need to check if the log file is already added to recycle list
+         * before we mark it for deletion.
+         */
+        if (!keep && (std::find(log_recycle_files.begin(),
+            log_recycle_files.end(),number) != log_recycle_files.end())) {
```

Comment:
We need to check this before putting in the full_scan_candidate_files then and we need to pase the file name for log number as well.
I saw PurgeObsoleteFiles for log files are called very infrequently though and probably adding mutex here will not be impacting performance ?